src/targets/awsLambdaLayer.ts

```diff
@@ -0,0 +1,202 @@
+import * as fs from 'fs';
+import { logger as loggerRaw } from '../logger';
+import { TargetConfig } from '../schemas/project_config';
+import { BaseTarget } from './base';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { ConfigurationError, reportError } from '../utils/errors';
+import {
+  AddLayerVersionPermissionRequest,
+  Lambda,
+  PublishLayerVersionCommandOutput,
+  PublishLayerVersionRequest,
+} from '@aws-sdk/client-lambda';
+import { DescribeRegionsCommandOutput, EC2 } from '@aws-sdk/client-ec2';
+
+const logger = loggerRaw.withScope(`[aws-lambda-layer]`);
+
+/** Config options for the "aws-lambda-layer" target. */
+interface AwsLambdaTargetOptions extends TargetConfig {
+  /** AWS access key ID, set as AWS_ACCESS_KEY_ID. */
+  awsAccessKeyId: string;
+  /** AWS secret access key, set as `AWS_SECRET_ACCESS_KEY`. */
+  awsSecretAccessKey: string;
+}
+
+/**
+ * Target responsible for uploading files to AWS Lambda.
+ */
+export class AwsLambdaLayerTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'aws-lambda-layer';
+  /** Target options */
+  public readonly awsLambdaConfig: AwsLambdaTargetOptions;
+  /** Name of the layer to be published */
+  static layerName: string;
+  /** All AWS regions available for the current user */
+  public awsRegions: any;
```

Comment:
```suggestion
  public awsRegions: string[];
```

?

Reply:
It looks like this could be simply a local variable in `publish` though, seems not to be used anywhere else.